### PR TITLE
Add support for option_support_large_channel

### PR DESCRIPTION
**Pull request by aarani**
This commit adds support for option_support_large_channel
and there are some changes made to validation process;
more specifically when validating our OpenChannel msg,
instead of just sending out an Error if the funding_satoshis
is greater than the limit, we check if support_large_channel
is activated as mandatory (cause if we don't set that option as
mandatory the openChannel will be invalid if sent to a peet with
no support for support_large_channel) and also when we are validating
someone else's openChannelMsg, we check that we have
support_large_channel option activated (it doesn't matter if mandatory
or optional).

### CODE REVIEW

**Comment by knocte** [comment]
typo in commit message: `It` -> `it`

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 1]
```diff
@@ -368,9 +369,17 @@ module internal OpenChannelMsgValidation =
...
+    let checkFundingSatoshisLessThanMax (msg: OpenChannelMsg) (localParams: LocalParams) (isOurOpenChannelMsg: bool) =
+        // If we are validating our own open message we make sure we are forcing support for option_support_large_channel on the other peer
+        let featureType =
+            match isOurOpenChannelMsg with
+            | true ->
+                Some FeaturesSupport.Mandatory
+            | false ->
+                None
+
+        if (msg.FundingSatoshis >= ChannelConstants.MAX_FUNDING_SATOSHIS && not (localParams.Features.HasFeature (Feature.OptionSupportLargeChannel, ?featureType = featureType))) then
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs:381]: 1. split this line, it's too long
2. now that you're changing it, let's remove the unneeded parentheses (C# needs them, F# doesn't)

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 2]
```diff
@@ -368,9 +369,17 @@ module internal OpenChannelMsgValidation =
...
+            sprintf "funding_satoshis must be less than %A. It was %A, consider activating option_support_large_channel feature." ChannelConstants.MAX_FUNDING_SATOSHIS msg.FundingSatoshis
```
Comment (knocte) [inline src/DotNetLightning.Core/Channel/ChannelError.fs:382]: let's split this line, it's too long

**Comment by knocte** [comment]
@joemphilips hey, we haven't had time to test this PR yet, but can you review? we'll test it after you've reviewed it

File: src/DotNetLightning.Core/Channel/ChannelError.fs [hunk 3]
```diff
@@ -368,9 +369,21 @@ module internal OpenChannelMsgValidation =
         else
             Ok()
 
-    let checkFundingSatoshisLessThanMax (msg: OpenChannelMsg) =
-        if (msg.FundingSatoshis >= ChannelConstants.MAX_FUNDING_SATOSHIS) then
-            sprintf "funding_satoshis must be less than %A. It was %A" ChannelConstants.MAX_FUNDING_SATOSHIS msg.FundingSatoshis
+    let checkFundingSatoshisLessThanMax (msg: OpenChannelMsg) (localParams: LocalParams) (isOurOpenChannelMsg: bool) =
```
Comment (joemphilips) [inline src/DotNetLightning.Core/Channel/ChannelError.fs:372]: By convention, OpenChannelMsg must be the last argument.

**Review by joemphilips — COMMENTED** [review]
LGTM besides one nit.

**Comment by aarani** [comment]
We tested this patch to create a 0.5 LTC channel and we can confirm it works.

**Comment by knocte** [comment]
We just tested this, it works, so it can be merged @joemphilips .

**Pull request merged**